### PR TITLE
fix: the nodes are parsed to the frontmatter even where they are not at the top.

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description
* frontmatter should be parsed only once from the top

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: libs/toastmark/src/commonmark/frontMatter/frontMatterStart.ts [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-      break;
-    }
-  }
-
-  if (canBeFrontmatter && !parser.indented && line.match(reFrontMatter)) {
+  if (
+    lineNumber === 1 &&
+    !indented &&
+    container.type === 'document' &&
+    reFrontMatter.test(currentLine)
```
Comment (jung-han) [inline libs/toastmark/src/commonmark/frontMatter/frontMatterStart.ts:14]: 해당 조건을 변수로 분리해서 의도가 드러나면 좀 더 좋을 것 같습니다!

**Review by jung-han — APPROVED** [review]
완료입니다!

**Review by lja1018 — APPROVED** [review]
리뷰완료합니다.

**Review by jajugoguma — APPROVED** [review]
리뷰 완료입니다!

**Review by dongsik-yoo — COMMENTED** [review]
리뷰 완료합니다. 수고하셨습니다.

**Review by adhrinae — APPROVED** [review]

**Pull request merged**